### PR TITLE
Completed sgt-back-end

### PR DESCRIPTION
**Pull request by Johnny-Baltazar**
get-response-200
get-response-500
post-response-201
post-missing-name
post-invalid-course
post-score-nan
post-score-101
post-response-500
put-response-200
put-invalid-name
put-missing-course
put-invalid-score
put-response-404
put-response-500
delete-response-204
delete-invalid-id
delete-response-404
delete-response-500

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An expected error occurred' });
+    });
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:33]: ✔

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  const postGrade = req.body;
+  const score = Number(postGrade.score);
+  if (!postGrade.name) {
+    res.status(400).json({ error: 'Must include a valid name' });
+  } else if (!postGrade.course) {
+    res.status(400).json({ error: 'Must include a valid course' });
+  } else if (!score) {
+    res.status(400).json({ error: 'Must include a valid score' });
+  } else if (score < 0 || score > 100 || isNaN(score)) {
+    res.status(400).json({ error: 'Score must be a positve integer between 0 & 100' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: While this catches non-numbers and numbers out of range, it doesn't catch non-integer numbers.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 2 above)
Comment (Johnny-Baltazar) [reply sgt-back-end/index.js]: put-not-integer
delete-not-integer

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        const grade = result.rows[0];
+        res.status(201);
+        res.json(grade);
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({ error: 'An unexpected error has occurred.' });
+      });
+  }
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:64]: ✔

File: sgt-back-end/index.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const gradeId = Number(req.params.gradeId);
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = req.body.score;
+
+  if (gradeId <= 0 || isNaN(gradeId)) {
+    res.status(400).send({ error: 'gradeId must be a postitve integer' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: `gradeId`s should also trigger a `400 Bad Request` if they are a non-integer number.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 4 above)
Comment (Johnny-Baltazar) [reply sgt-back-end/index.js]: gradeId-not-integer

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**